### PR TITLE
Add todolist example

### PR DESCRIPTION
**Pull request by danielart**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[ x] Other, please explain: Add todolist example of use

**What changes did you make? (Give an overview)**
add a new folder of examples, with a todo list folder (first example)

**Which issue (if any) does this pull request address?**

**Is there anything you'd like reviewers to focus on?**
please check the structure

### CODE REVIEW

File: examples/todo-list/src/AddTodoTask.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useState } from "react";
+import { useStore } from "./store";
+import { FcTodoList } from "react-icons/fc";
```
Comment (aralroca) [inline examples/todo-list/src/AddTodoTask.js:3]: Is necessary this import?

File: examples/todo-list/src/AddTodoTask.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export default function AddTodoTask() {
+  const [counter, setCounter] = useState(0);
```
Comment (aralroca) [inline examples/todo-list/src/AddTodoTask.js:6]: I feel this counter inconsistent, navegating to another page + come back (unmount + mount) is going to repeat id's...

File: examples/todo-list/src/AddTodoTask.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    console.log(JSON.stringify(e.target.children[0].value));
+    e.preventDefault();
+    if (e.target.children[0].value.trim() === "") {
+      setError("Can't add empty tasks");
+      return;
+    } else setError(undefined);
+
+    setTodo({
+      ...todo,
+      [counter]: { text: e.target.children[0].value, id: counter }
```
Comment (aralroca) [inline examples/todo-list/src/AddTodoTask.js:19]: Maybe using an epoch for id's is enough

```suggestion
      [counter]: { text: e.target.children[0].value, id: Date.now() }
```

File: examples/todo-list/src/AddTodoTask.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    });
+    e.target.children[0].value = "";
+    setCounter(counter + 1);
```
Comment (aralroca) [inline examples/todo-list/src/AddTodoTask.js:22]:
```suggestion
```

**Review by aralroca — COMMENTED** [review]
The test of the example is executed as repo test and fails for some dependencies. What I would do is have `yarn test` only to run the repo tests and then `yarn test:examples` for all the tests in the examples.

File: examples/todo-list/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "@testing-library/dom": "8.10.1",
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "12.1.2",
+    "@testing-library/user-event": "13.5.0",
+    "fragstore": "0.4.1",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
+    "react-icons": "^4.3.1",
+    "react-scripts": "4.0.0",
+    "teaful": "^0.7.0"
```
Comment (aralroca) [inline examples/todo-list/package.json:17]:
```suggestion
    "teaful": "link:../"
```

**Comment by aralroca** [comment]
@danielart if you add this in [`package.json`](https://github.com/teafuljs/teaful/blob/0.7.2/package.json#L67-L69) fix the repo tests:

https://github.com/teafuljs/teaful/blob/0.7.2/package.json#L67-L69

```diff
"jest": {
    "testEnvironment": "jsdom",
+    "testPathIgnorePatterns": ["/node_modules/", "/examples/"]
  }
```

File: examples/todo-list/package.json (hunk 5 above)
Comment (danielart) [reply examples/todo-list/package.json:17]: A we talked about this, we keep a fixed version of teaful at the moment to test manually  this examples when version is updated to be sure all is working fine

File: examples/todo-list/src/AddTodoTask.js (hunk 1 above)
Comment (danielart) [reply examples/todo-list/src/AddTodoTask.js:3]: deleted, was from a deleted icon that we're not using anymore

File: examples/todo-list/src/AddTodoTask.js (hunk 2 above)
Comment (danielart) [reply examples/todo-list/src/AddTodoTask.js:5]: having in mind this todo app is not growing and there is no plan to change page or unmount the component, I think we could leave as is

**Comment by danielart** [comment]
> @danielart if you add this in [`package.json`](https://github.com/teafuljs/teaful/blob/0.7.2/package.json#L67-L69) fix the repo tests:
> 
> https://github.com/teafuljs/teaful/blob/0.7.2/package.json#L67-L69
> 
> ```diff
> "jest": {
>     "testEnvironment": "jsdom",
> +    "testPathIgnorePatterns": ["/node_modules/", "/examples/"]
>   }
> ```
> 
> Then I would add an extra script: `yarn test:examples`. Maybe we can use [yarn workspaces](https://yarnpkg.com/features/workspaces) to manage these tests. If you find it difficult we can do a pair programming and we can look at it together.

I splitted the test and format scripts, test:examples and format:examples (this last one to have the same eslint on examples, but avoid to spend time formatting examples when not necessary)

I added an extra config  for jest on package.json, but If you think could be better with yarn workspaces we can check it out

**Review by aralroca — APPROVED** [review]

**Comment by aralroca** [comment]
@all-contributors please add @danielart for code

**Pull request merged**